examples/server.rs

```diff
@@ -0,0 +1,159 @@
+#[derive(StructOpt, Debug)]
+#[structopt(name = "server")]
+struct Opt {
+    #[structopt(subcommand)]
+    pub command: Command,
+}
+
+#[derive(StructOpt, Debug)]
+pub enum Command {
+    #[structopt(name = "selfsigned")]
+    SelfSigned(SelfSigned),
+
+    #[structopt(name = "certs")]
+    Certs(Certs),
+}
+
+#[derive(StructOpt, Debug)]
+pub struct SelfSigned {
+    #[structopt(long)]
+    pub debug: bool,
+
+    #[structopt(long, default_value = "4433")]
+    pub port: u16,
+}
+
+#[derive(StructOpt, Debug)]
+pub struct Certs {
+    #[structopt(long)]
+    pub cert: PathBuf,
+
+    #[structopt(long)]
+    pub key: PathBuf,
+
+    #[structopt(long, default_value = "4433")]
+    pub port: u16,
+}
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    tracing_subscriber::fmt()
+        .with_env_filter(tracing_subscriber::EnvFilter::from_default_env())
+        .with_span_events(tracing_subscriber::fmt::format::FmtSpan::FULL)
+        .with_writer(std::io::stderr)
+        .init();
+
+    let opt = Opt::from_args();
+    trace!("{:#?}", opt);
+
+    // quinn setup
+    let mut server_config = h3_quinn::quinn::ServerConfigBuilder::default();
+    server_config.protocols(&[b"h3-27"]);
+
+    let (endpoint, mut incoming) = match opt.command {
+        Command::SelfSigned(r) => {
+            let (cert_chain, _cert, key) = build_certs();
+
+            server_config.certificate(cert_chain, key).unwrap();
+
+            let mut server_endpoint_builder = h3_quinn::quinn::Endpoint::builder();
+            server_endpoint_builder.listen(server_config.build());
+
+            let addr = format!("[::]:{:}", r.port);
+
+            server_endpoint_builder
+                .bind(&addr.parse().unwrap())
+                .unwrap()
+        }
+        Command::Certs(c) => {
+            let mut cert_v = Vec::new();
+            let mut key_v = Vec::new();
+
+            let mut cert_f = File::open(c.cert).await?;
+            let mut key_f = File::open(c.key).await?;
+
+            cert_f.read_to_end(&mut cert_v).await?;
+            key_f.read_to_end(&mut key_v).await?;
+
+            server_config
+                .certificate(
+                    h3_quinn::quinn::CertificateChain::from_pem(cert_v.as_slice())?,
+                    h3_quinn::quinn::PrivateKey::from_pem(key_v.as_slice())?,
+                )
+                .unwrap();
+
+            let mut server_endpoint_builder = h3_quinn::quinn::Endpoint::builder();
+            server_endpoint_builder.listen(server_config.build());
+
+            let addr = format!("[::]:{:}", c.port);
+
+            server_endpoint_builder
+                .bind(&addr.parse().unwrap())
+                .unwrap()
+        }
+    };
+
+    warn!(
+        "Listening on port {:?}",
+        endpoint.local_addr().unwrap().port()
+    );
+
+    while let Some(new_conn) = incoming.next().await {
+        trace_span!("New connection being attempted");
```

Comment:
So that *this* connection doesn't block others, this should probably spawn a new task, `tokio::spawn`.